frontend/src/metabase/search/components/SearchSidebar/SearchSidebar.tsx

```diff
@@ -1,59 +1,104 @@
 import _ from "underscore";
+import { useCallback, useMemo } from "react";
 import type {
   FilterTypeKeys,
+  SearchFilterComponent,
   SearchFilterPropTypes,
   URLSearchFilterQueryParams,
-  SearchSidebarFilterComponent,
 } from "metabase/search/types";
 import { Stack } from "metabase/ui";
 import { SearchFilterKeys } from "metabase/search/constants";
-import { SidebarFilter } from "metabase/search/components/SidebarFilter/SidebarFilter";
+import { DropdownSidebarFilter } from "metabase/search/components/SearchSidebar/DropdownSidebarFilter/DropdownSidebarFilter";
 import { TypeFilter } from "metabase/search/components/filters/TypeFilter/TypeFilter";
+import { PLUGIN_CONTENT_VERIFICATION } from "metabase/plugins";
+import { ToggleSidebarFilter } from "metabase/search/components/SearchSidebar/ToggleSidebarFilter/ToggleSidebarFilter";
 import { CreatedByFilter } from "metabase/search/components/filters/CreatedByFilter/CreatedByFilter";
 
 type SearchSidebarProps = {
   value: URLSearchFilterQueryParams;
   onChange: (value: URLSearchFilterQueryParams) => void;
 };
 
-export const filterMap: Record<FilterTypeKeys, SearchSidebarFilterComponent> = {
-  [SearchFilterKeys.Type]: TypeFilter,
-  [SearchFilterKeys.CreatedBy]: CreatedByFilter,
-};
-
 export const SearchSidebar = ({ value, onChange }: SearchSidebarProps) => {
+  const filterMap: Record<FilterTypeKeys, SearchFilterComponent | null> =
+    useMemo(
+      () => ({
+        [SearchFilterKeys.Type]: TypeFilter,
+        [SearchFilterKeys.CreatedBy]: CreatedByFilter,
+        [SearchFilterKeys.Verified]: PLUGIN_CONTENT_VERIFICATION.VerifiedFilter,
+      }),
+      [],
+    );
+
+  const isValidFilterValue = useCallback(
+    (
+      key: FilterTypeKeys,
+      val: SearchFilterPropTypes[FilterTypeKeys],
+    ): boolean => {
+      if (!val || !filterMap[key]) {
+        return false;
+      }
+
+      if (Array.isArray(val)) {
+        return val.length > 0;
+      }
+      return true;
+    },
+    [filterMap],
+  );
+
   const onOutputChange = (
     key: FilterTypeKeys,
     val: SearchFilterPropTypes[FilterTypeKeys],
   ) => {
-    if (!val || (Array.isArray(val) && val.length === 0)) {
+    if (!isValidFilterValue(key, val)) {
       onChange(_.omit(value, key));
     } else {
-      const { toUrl } = filterMap[key];
+      const filterMapElement = filterMap[key];
+      const toUrl = filterMapElement?.toUrl;
```

Comment:
Is this still needed even though we filter out invalid filter values on line 54? Feels to me `filterMapElement` at this point shouldn't be nullable.

Or is this change needed just to satisfy TypeScript? If that's the case, I think we should fix `isValidFilterValue` to let TypeScript know it's filtering out invalid values instead.